src/views/newConnection.tsx

```diff
@@ -1,42 +1,118 @@
+        <form
+          id="new-connection-form"
+          class="new-connection"
+          hx-post="/connection/new/submit"
+          hx-swap="outerHTML"
+          hx-vals={`{"formStage": "${params.formStage}"}`}
+        >
+          <div class="new-connection col">
+            <this.stepper formStage={params.formStage} />
+            <input
+              id="company-number-input"
+              class="new connection"
+              name="companyNumber"
+              placeholder="Company House Number"
+              required
+              hx-get="/connection/new/verify-company"
+              hx-trigger="keyup changed delay:200ms, change"
+              hx-target="#text-box-target"
+              hx-select="#text-box-target"
+              pattern="^(((AC|CE|CS|FC|FE|GE|GS|IC|LP|NC|NF|NI|NL|NO|NP|OC|OE|PC|R0|RC|SA|SC|SE|SF|SG|SI|SL|SO|SR|SZ|ZC|\d{2})\d{6})|((IP|SP|RS)[A-Z\d]{6})|(SL\d{5}[\dA]))$"
+              minlength={8}
+              maxlength={8}
+              oninput="this.reportValidity()"
+              value={params.companyNumber}
+              type={`${showForm ? 'text' : 'hidden'}`}
+            ></input>
+            <input
+              id="email"
+              class="new-connection"
+              placeholder="Connection's Email Address"
+              name="email"
+              value={params.email}
+              type={`${showForm ? 'email' : 'hidden'}`}
+            ></input>
+            <div id="confirmation-page" style={{ display: showConfirmation ? 'block' : 'none' }}>
+              <p>Please confirm the details of the connection before sending</p>
+              <p>
+                {Html.escapeHtml(
+                  `Company House Number: ${params.targetBox.status === 'success' && params.targetBox.company.company_number}`
+                )}
+              </p>
+              <p>{Html.escapeHtml(`Email Address: ${params.targetBox.status === 'success' && params.email}`)}</p>
+            </div>
+            <div id="success-page" style={{ display: showSuccess ? 'block' : 'none' }}>
+              <p>
+                Your connection invitation has been sent. Please wait for their verification. As the post may take 2-3
+                days to arrive, please wait for their verification and keep updated by viewing the verification status.
+              </p>
+            </div>
+            <a class="button" href="/connection" style={{ display: showConfirmation ? 'none' : 'block' }}>
+              {showForm ? 'Cancel' : 'Back To Home'}
+            </a>
+            <FormButtonIcon
+              type="submit"
+              name="submitButton"
+              value="Back"
+              display={showConfirmation ? 'block' : 'none'}
+            />
+            <FormButtonIcon
+              type="submit"
+              name="submitButton"
+              value={`${showForm ? 'Continue' : 'Submit'}`}
+              display={showSuccess ? 'none' : 'block'}
+            />
```

Comment:
Since it's same type and same name. maybe we could combine into a single one as per below

```js
            <FormButtonIcon
              type="submit"
              name="submitButton"
              value={showConfirmation ? "Back" : showSuccess ? "Continue" : "Submit"}
            />
```
Not sure about the typing might need to double check, but this way we do not need to use "display" additional <FormButtonIcon />

Reply:
So the two `<FormButtonIcon>` will be shown at the same time in some of the views